plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/expression/RewriteLike.java

```diff
@@ -0,0 +1,102 @@
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RewriteLike
+        implements ConnectorExpressionRule<Call, ParameterizedExpression>
+{
+    private static final Capture<Variable> LIKE_VALUE = newCapture();
+    // TODO allow Variable as a LIKE_PATTERN: "SELECT * FROM t WHERE column_a LIKE column_b" is a valid query in ClickHouse
+    // only Constant is allowed as LIKE_PATTERN, because according to
+    // https://clickhouse.com/docs/en/sql-reference/functions/string-search-functions#like
+    // ClickHouse requires backslashes in strings to be quoted as well, so you would actually need to write \\%, \\_ and \\\\ to match against literal %, _ and \
+    // if "column_a LIKE column_b" is pushed down, it requires more thorough consideration how to process escaping.
+    private static final Capture<Constant> LIKE_PATTERN = newCapture();
+    private static final Pattern<Call> PATTERN = call()
+            .with(functionName().equalTo(LIKE_FUNCTION_NAME))
+            .with(type().equalTo(BOOLEAN))
+            .with(argumentCount().equalTo(2))
+            .with(argument(0).matching(variable()
+                    .with(type().matching(type -> type instanceof CharType || type instanceof VarcharType))
+                    .matching((Variable variable, RewriteContext<ParameterizedExpression> context) -> supportsPushdown(variable, context))
+                    .capturedAs(LIKE_VALUE)))
+            .with(argument(1).matching(constant()
+                    .with(type().matching(type -> type instanceof CharType || type instanceof VarcharType))
+                    .capturedAs(LIKE_PATTERN)));
+
+    @Override
+    public Pattern<Call> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Optional<ParameterizedExpression> rewrite(Call expression, Captures captures, RewriteContext<ParameterizedExpression> context)
+    {
+        Optional<ParameterizedExpression> value = context.defaultRewrite(captures.get(LIKE_VALUE));
+        if (value.isEmpty()) {
+            return Optional.empty();
+        }
+        Optional<ParameterizedExpression> pattern = context.defaultRewrite(captures.get(LIKE_PATTERN));
+        if (pattern.isEmpty()) {
+            return Optional.empty();
+        }
+
+        List<QueryParameter> patternParameters = pattern.get().parameters();
+        if (patternParameters.size() != 1
+                || patternParameters.getFirst().getValue().isEmpty()
+                || !(patternParameters.getFirst().getType() instanceof VarcharType || patternParameters.getFirst().getType() instanceof CharType)) {
+            return Optional.empty();
+        }
+
+        Slice slice = (Slice) patternParameters.getFirst().getValue().get();
+        String parameter = new String(slice.byteArray(), UTF_8)
+                // ClickHouse requires backslashes in strings to be quoted as well, so you would actually need to write \\%, \\_ and \\\\ to match against literal %, _ and \
```

Comment:
This should happen only if we have the Escape clause right ?

Reply:
As you noted, LIKE in trino does not support escape characters.
However Clickhouse supports them with
> `\` is for escaping literals `%`, `_` and `\`.

So just to translate Trino `LIKE '\'`  , all `\` have to be escaped, and resulting clickhouse should be `LIKE '\\\\'`

Reply:
For now disabling pushdown when pattern contains `\`.